src/com/github/korniloval/matlab/editor/smartEnter/StatementCompletionProcessor.kt

```diff
@@ -0,0 +1,64 @@
+package com.github.korniloval.matlab.editor.smartEnter
+
+import com.github.korniloval.matlab.psi.MatlabAssignExpr
+import com.github.korniloval.matlab.psi.MatlabFunctionExpr
+import com.intellij.codeInsight.editorActions.smartEnter.SmartEnterProcessor
+import com.intellij.openapi.editor.Editor
+import com.intellij.openapi.project.Project
+import com.intellij.openapi.util.TextRange
+import com.intellij.openapi.util.text.StringUtil
+import com.intellij.psi.PsiElement
+import com.intellij.psi.PsiFile
+import java.util.*
+import java.util.Optional.*
+
+class StatementCompletionProcessor : SmartEnterProcessor() {
+
+  override fun process(project: Project, editor: Editor, psiFile: PsiFile): Boolean {
+    return ofNullable(getStatementAtCaret(editor, psiFile))
+        .flatMap { element ->
+          findStatement(element) { node ->
+            when (node) {
+              is MatlabFunctionExpr,
+              is MatlabAssignExpr -> true
+              else -> false
+            }
+          }
+        }
+        .map { statementElement ->
+          val statementRange = getStatementRange(editor, statementElement)
+          val fullStatement = editor.document.getText(statementRange)
+          val endOfStatement =
+              completeStatementIfNecessary(fullStatement, editor, statementElement.textRange.endOffset)
+          editor.caretModel.moveToOffset(endOfStatement + 1)
```

Comment:
It should be `editor.caretModel.moveToOffset(endOfStatement)` because we do not want to move caret to the very beginning of new line

Reply:
Yes, this is how it should be. To achieve this we need to return `false` if `;` is present (in this case new line will be inserted automatically)

Reply:
See [SmartEnterAction:101](https://github.com/JetBrains/intellij-community/blob/master/platform/lang-impl/src/com/intellij/codeInsight/editorActions/smartEnter/SmartEnterAction.java#L101)